### PR TITLE
Enable Travis CI to build and test our non-matlab components

### LINKED ISSUE #1229: Consider using TravisCI as an additional test system
If we're going to make Drake work well without Matlab, then we're going to need a non-matlab build environment to test with. While I'm sure we could just set up yet another build server, I'd suggest we consider using  a free service like Travis https://travis-ci.org/ to handle the build and test process for us. I've been using it with IRIS and other projects, and I've found it really easy to work with. 

Here are a few advantages I see:
* It automatically integrates with github pull requests to show you the status of the build and test for each PR. For example: https://github.com/rdeits/iris-distro/pull/21 (and it knows to re-run the tests if any new commits are added to the PR). 
* It also doesn't spam the github comments with build output :wink: 
* The entire build server is spun up as a new virtual machine for every build, so there's no weird state persistence (although it's also possible to cache slow parts of the build if you want)
* It's free (for public repos) and open source (but you can also pay them for private repos)
* It allows multiple build configurations. For example, I've set up IRIS to test every commit with both `gcc` and `clang` as separate builds. 

And some disadvantages:
* It doesn't give a cdash-like display of the result of every test separately. Instead, it just reports success or failure for the whole build process. That means that if we want to handle "known failures" in the c++ tests, we have to write that logic ourselves. 
* It's another thing to understand and configure. 

Fortunately, I think the configuration is pretty easy. I'm working on getting drake to build on Travis right now from my fork, and the config script is just:

```yaml
language: cpp

sudo: required

notification:
  email: false

compiler:
  - gcc
  - clang

addons:
  apt:
    sources:
      - ubuntu-toolchain-r-test
    packages:
      - gcc-4.8
      - g++-4.8

install:
  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8" CC="gcc-4.8"; fi

before_install:
  - make download-all
  - sudo ./install_prereqs.sh ubuntu

script:
  - mkdir build
  - make
  - CTEST_OUTPUT_ON_FAILURE=1 make test
```

### LINKED ISSUE #1230: Problems with install_prereqs.sh
I've found a couple of problems with install_prereqs.sh:

1. Since it invokes `apt-get install`, it must be run as root (on ubuntu). But it also calls `make download-all`, which results in all of the external source folders being owned by root (which is bad).
1. Because it calls the Makefile, it invokes cmake and ends up setting up some initial Cmake cache variables. That's a problem because it always sets the BUILD_TYPE to "Release", which makes a debug build impossible.

### PR DESCRIPTION
**Pull request by rdeits**
This branch creates the `.travis.yml` file, which contains the configuration options to build and test Drake with Travis: https://travis-ci.org/

I've also fixed a few build issues that came up when building without matlab installed. We can now run 11 thrilling c++-only unit tests :wink: 

As a demo, I've set up the Travis build to run 4 sets of tests:
* gcc + Release
* gcc + Debug
* clang + Release
* clang + Debug

We can reconfigure the set of tests that get run however we want. 

I've got Travis building on my drake fork right now, and you can see what the results look like: https://travis-ci.org/rdeits/drake

closes #1230 
closes #1229

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,8 +103,8 @@ set(bertini_GIT_REPOSITORY https://github.com/RobotLocomotion/bertini.git)
 set(bertini_GIT_TAG 3ae3e3ad3534acb4b6f7a4c3c22728b493beaa80)
 set(bullet_GIT_REPOSITORY https://github.com/RobotLocomotion/bullet-pod.git)
 set(bullet_GIT_TAG 6db7ae5e29f8b35ffcea384c53caa4bddcc6f986)
-set(cmake_GIT_REPOSITORY https://github.com/RobotLocomotion/cmake.git)
-set(cmake_GIT_TAG c48d1e461d8f0a8d67ebba3e4acb19d96dd5a25b)
+set(cmake_GIT_REPOSITORY https://github.com/rdeits/cmake.git)
```
Comment (rdeits) [inline CMakeLists.txt]: once again temporarily pointing this to my fork for testing

**Comment by rdeits** [comment]
I'm not sure what that last error on drake006 was...I'm going to close and re-open to try again

**Comment by rdeits** [comment]
No response from drake006. I guess I'll bump this again?

**Comment by rdeits** [comment]
drake001 looks like it failed with the lcm java permissions issue. I'm not sure what happened to drake003. @RussTedrake can you re-trigger those?

**Comment by rdeits** [comment]
drake003 is failing all over the place because of a permissions issue with `lcppath.mexw64`, but I'm not sure what's going on. The file is there and is definitely readable. Weird...

**Comment by rdeits** [comment]
Oh, I see that's a known issue, never mind. There are 4 unknown failures on drake003, though, and they've all got error reports which are cut off. I'll try to find that build and see what went wrong.

**Comment by rdeits** [comment]
After a while spent poking around on drake003 I still have no idea what happened. Some of the c++ IK tests are segfaulting, but I can't reproduce them without figuring out how `ld_library_path` works on cygwin.

**Comment by RussTedrake** [comment]
It's known.  Path doesn't work on windows.

**Comment by RussTedrake** [comment]
The link to pat's trigpoly.h pull request drake003 results actually showed the errors (the first time i've seen the test output!). Those tests aren't finding the atlas files because of a ros package path issue...

**Comment by patmarion** [comment]
Hey Robin, I just had this build error when matlab wasn't in the PATH.  I assume this branches fixes the issue and I don't need to open a new issue?

-- lqrmex will be skipped because matlab was not found
CMake Error at solvers/CMakeLists.txt:31 (target_link_libraries):
  Cannot specify link libraries for target "lqrmex" which is not built by
  this project.

**Comment by rdeits** [comment]
@patmarion yeah, it should be fixed. Travis is building with no matlab at all.

**Comment by rdeits** [comment]
I'm afraid I'm starting to lose track of exactly which commits are where, though, since #1223 is also based on this branch, but has some other weird new failures.

**Comment by rdeits** [comment]
Ok, I think this one's finally ready to go?

**Comment by rdeits** [comment]
Woohoo! Okay @RussTedrake if you flip the switch on https://travis-ci.org/RobotLocomotion/drake we should start getting build results right away

File: CMakeLists.txt (hunk 1 above)
Comment (RussTedrake) [inline CMakeLists.txt]: Doh. forgot you still had this pointing here.

**Comment by RussTedrake** [comment]
Will do tonight (am on the phone right now)

File: CMakeLists.txt (hunk 1 above)
Comment (rdeits) [inline CMakeLists.txt]: ah, d'oh. Fixed in #1249

**Pull request merged**